### DISCUSSION TITLE
6.0.0 FVTT v10.284 Compatibility || Damage Console

### DISCUSSION THREAD

**Discussion by Jagusti** [Jagusti/fvtt-wfrp4e-gmtoolkit#174] [Announcements]
## Release Summary

This release confirms Foundry v10 compatibility, building on changes introduced in [Release Candidate v0.9.5](https://github.com/Jagusti/fvtt-wfrp4e-gmtoolkit/releases/tag/v0.9.5), and as defined in issue #156 .  

**Requires** at least Foundry v10.284 and WFRP 6.1.2. 

### Feature Updates

In addition to v10 compatibility, this release: 

* _Adds_ the [Damage Console](https://github.com/Jagusti/fvtt-wfrp4e-gmtoolkit/pull/168), from which GMs can quickly set and apply damage to one or more actors without starting a combat, opposed or other test. See [below](#damage-console) for key features. #158 
* _Fixes_ an issue where a combatant could gain Advantage from an Opposed Test they won but did not initiate when Group Advantage is being used. #159 
* _Changes_ Group Test behaviour to clear or restore the custom skill field if a skill is chosen from the skill list dropdown. #160 

### Repository Management

Module management and repository changes have also been made:
* _Version numbering_ is overhauled, essentially to keep in step with WFRP4e system versioning. 
  * This is marked by a jump from v0.* to 6.*. 
  * There is no v1-v5 of GM Toolkit. 
* _Templates_ are provided for [bug reports](https://github.com/Jagusti/fvtt-wfrp4e-gmtoolkit/blob/dev/.github/ISSUE_TEMPLATE/issue_report.md), [feature suggestions](https://github.com/Jagusti/fvtt-wfrp4e-gmtoolkit/blob/dev/.github/ISSUE_TEMPLATE/feature_request.md) and [pull requests](https://github.com/Jagusti/fvtt-wfrp4e-gmtoolkit/blob/dev/.github/PULL_REQUEST_TEMPLATE/pull_request_template.md).  
* _[Discussions](https://github.com/Jagusti/fvtt-wfrp4e-gmtoolkit/discussions)_ are enabled on the repository. I intend to use these to solicit input and feedback on new features, enhancements and potential bugs. 

## Damage Console
Issue: #158. Pull Request: #168.

* Access from the **Launch Damage Console** macro or the **GM Toolbox**.
* Target tokens before launching to change the pre-selected characters. 
* Set damage as a roll formula or fixed amount.
* Optionally randomise damage and hit location for each target. 
* Choose whether to ignore Armour Points and/or Toughness Bonus, and whether to apply a minimum of 1 damage.
* Check tooltips to find randomisation details.

![image](https://user-images.githubusercontent.com/521096/188506217-de95975c-347e-469a-bb7a-f288d2c71a66.png)
![image](https://user-images.githubusercontent.com/521096/188506326-d527df6c-a6a5-4109-a93c-28d34d498c6d.png)

## What's Changed
* Add compatibility for Foundry VTT v10  and WFRP4e v6 https://github.com/Jagusti/fvtt-wfrp4e-gmtoolkit/pull/162
* Add templates for issues and PRs https://github.com/Jagusti/fvtt-wfrp4e-gmtoolkit/pull/163 https://github.com/Jagusti/fvtt-wfrp4e-gmtoolkit/pull/164 https://github.com/Jagusti/fvtt-wfrp4e-gmtoolkit/pull/171
* Clear or restore default Group Test skill #160 https://github.com/Jagusti/fvtt-wfrp4e-gmtoolkit/pull/166
* Prevent advantage gain if not initiating challenge #159 https://github.com/Jagusti/fvtt-wfrp4e-gmtoolkit/pull/167
* Add Damage Console #158 https://github.com/Jagusti/fvtt-wfrp4e-gmtoolkit/pull/168
* Add release workflow https://github.com/Jagusti/fvtt-wfrp4e-gmtoolkit/pull/169
* 6.0.0 macro versioning https://github.com/Jagusti/fvtt-wfrp4e-gmtoolkit/pull/173

**Full Changelog**: https://github.com/Jagusti/fvtt-wfrp4e-gmtoolkit/compare/v0.9.4.4...6.0.0

<hr /><em>This discussion was created from the release <a href='https://github.com/Jagusti/fvtt-wfrp4e-gmtoolkit/releases/tag/6.0.0'>6.0.0 FVTT v10.284 Compatibility || Damage Console</a>.</em>

**Comment by nyoung** [comment]
I really like the damage console, but its picking up every Actor that is assigned Owner permissions, which is including horses, loot, etc, which gives me a fairly lenghty list to go through to find the characters impacted. Is there a way to limit it to the characters in the current scene or to characters actually assigned to the player?

**Reply by Jagusti** [reply]
Thanks for the appreciation :) 
By default the **damage console** builds the player's character list from the `entourage` group type. At the moment this is baked in and can't be customised without changing the source code. 

If you want to change it to a different group type, you can adjust [this line in /apps/damage.js:](https://github.com/Jagusti/fvtt-wfrp4e-gmtoolkit/blob/8a0e5ea86deb0c769cba71fc419cb544103b2985/apps/damage.js#L33):
```
      playerGroup: game.gmtoolkit.utility.getGroup("entourage"),
```

Change `entourage` to: 
- `pcTokens` to limit to those PCs represented by tokens in the scene
- `party` for just the assigned PCs (regardless of whether they have  a token in the scene)
- `company` for player-owned, non-vehicle actors (whether in the scene or not)

Let me know how that goes for you. Also, it would be great to understand the sort of numbers you're talking about, and how it renders. If you can share some more info and a screenshot I'd apprecaite that - it will help with understanding real use cases and inform the sort of options I might be able to offer in the future. 

One likely future option will be to choose the default Damage group, similar to how you can choose party or company for Group Tests.
